src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunMinimalBWAMEM.java

```diff
@@ -0,0 +1,193 @@
+     *          empty if caller decides to redirect the stderr message to a file
+     */
+    @Override
+    public String doWork(){
+
+        validateUserOptions();
+
+        final BWAMEMModule bwamem = new BWAMEMModule();
+        final Tuple3<Integer, String, String> result = bwamem.run(Paths.get(pathToBWA), new File(outputDir), makeArgs());
+
+        return validateResults(result);
+    }
+
+    private void validateUserOptions() throws UserException {
+        final boolean validOptions = (null==secondInput) ? !(interLeaved && SEInput) : (!interLeaved || SEInput);
+        if(!validOptions){
+            throw new UserException("CMD line argument options on input (paired, interleaved, or SE) don't make sense. Please check.");
+        }
+    }
+
+    private String validateResults(final Tuple3<Integer, String, String> result) throws GATKException {
+
+        switch (result._1()){
+            case 0:  return writeSamFile(result);
+            case 1:  throw new GATKException("Failed to start bwa mem." + System.getProperty("line.separator") + result._3());
+            case 2:  throw new GATKException("The bwa mem process was interrupted." + System.getProperty("line.separator") + result._3());
+            case 3:  throw new GATKException("Failed to capture bwa stdout/stderr message" + System.getProperty("line.separator") + result._3());
+            default: throw new GATKException("bwa mem returned non-zero code: " + Integer.toString(result._1() - 10) +
+                                                System.getProperty("line.separator") + result._3());
+        }
+    }
+
+    /**
+     * Writes stdout from bwa mem to designated file.
+     * Return stderr message as string, which will be empty if user decides to redirect stderr message to file (i.e. stderrDestFileName is set)
+     */
+    private String writeSamFile(final Tuple3<Integer, String, String> result){
+
+        try{
+            final File wkDir = new File(outputDir);
+            final File samFile = new File(wkDir, samOutput);
+            FileUtils.writeStringToFile(samFile, result._2());
+
+        } catch (final IOException e){
+            throw new GATKException("Failed to dump bwa mem stdout to designated sam file." +
+                                        System.getProperty("line.separator") + e.getMessage());
+        }
+
+        if(null==stderrDestFileName){
+            return result._3();
+        }else{
+            final File wkDir = new File(outputDir);
+            final File stderrFile = new File(wkDir, stderrDestFileName);
+            try{
+                FileUtils.writeStringToFile(stderrFile, result._3());
+            }catch(final IOException e){
+                System.err.println("Failed to dump stderr message form bwa mem to designated file:" + System.getProperty("line.separator") +
+                                    outputDir + System.getProperty("line.separator") + stderrDestFileName + System.getProperty("line.separator"));
```

Comment:
don't bother with `System.getProperty("line.separator")` - just use "\n". We have this already everywhere. Or is you really want, use a constant and dont repeat `System.getProperty("line.separator")` everywhere here